openmc/mgxs_library.py

```diff
@@ -0,0 +1,2577 @@
+    xs_shapes : dict of iterable of int
+        Dictionary with keys of _XS_SHAPES and iterable of int values with the
+        corresponding shapes where "Order" corresponds to the pn scattering
+        order, "G" corresponds to incoming energy group, "G'" corresponds to
+        outgoing energy group, and "DG" corresponds to delayed group.
+
+    Notes
+    -----
+    The parameters containing cross section data have dimensionalities which
+    depend upon the value of :attr:`XSdata.representation` as well as the
+    number of Legendre or other angular dimensions as described by
+    :attr:`XSdata.order`. The :attr:`XSdata.xs_shapes` are provided to obtain
+    the dimensionality of the data for each temperature.
+
+    The following are cross sections which should use each of the properties.
+    Note that some cross sections can be input in more than one shape so they
+    are listed multiple times:
+
+    [G][G'][Order]: scatter_matrix
+
+    [G]: total, absorption, fission, kappa_fission, nu_fission,
+         prompt_nu_fission, delayed_nu_fission, inverse_velocity
+
+    [G']: chi, chi_prompt, chi_delayed
+
+    [G][G']: multiplicity_matrix, nu_fission, prompt_nu_fission
+
+    [DG]: beta, decay_rate
+
+    [DG][G]: delayed_nu_fission, beta, decay_rate
+
+    [DG][G']: chi_delayed
+
+    [DG][G][G']: delayed_nu_fission
+
+    """
+
+    def __init__(self, name, energy_groups, temperatures=[294.],
+                 representation='isotropic', num_delayed_groups=0):
+
+        # Initialize class attributes
+        self.name = name
+        self.energy_groups = energy_groups
+        self.num_delayed_groups = num_delayed_groups
+        self.temperatures = temperatures
+        self.representation = representation
+        self._atomic_weight_ratio = None
+        self._fissionable = False
+        self._scatter_format = 'legendre'
+        self._order = None
+        self._num_polar = None
+        self._num_azimuthal = None
+        self._total = len(temperatures) * [None]
+        self._absorption = len(temperatures) * [None]
+        self._scatter_matrix = len(temperatures) * [None]
+        self._multiplicity_matrix = len(temperatures) * [None]
+        self._fission = len(temperatures) * [None]
+        self._nu_fission = len(temperatures) * [None]
+        self._prompt_nu_fission = len(temperatures) * [None]
+        self._delayed_nu_fission = len(temperatures) * [None]
+        self._kappa_fission = len(temperatures) * [None]
+        self._chi = len(temperatures) * [None]
+        self._chi_prompt = len(temperatures) * [None]
+        self._chi_delayed = len(temperatures) * [None]
+        self._beta = len(temperatures) * [None]
+        self._decay_rate = len(temperatures) * [None]
+        self._inverse_velocity = len(temperatures) * [None]
+        self._xs_shapes = None
+
+    def __deepcopy__(self, memo):
+        existing = memo.get(id(self))
+
+        # If this is the first time we have tried to copy this object, copy it
```

Comment:
It might be fine for this scenario, but it seems worth noting that any updates made to the object after the first copy won't show up if copied again. Instead, the first copy will be used.